### PR TITLE
feat(utxo-lib): support p2trMusig2 key path sig verification

### PR DESCRIPTION
**Pull request by saravanan7mani**
Verify signature for p2trMusig2 key path PSBT input.

TICKET: BG-66935

### CODE REVIEW

File: modules/utxo-lib/src/bitgo/Musig2.ts [hunk 1]
```diff
@@ -242,14 +257,38 @@ export function musig2PartialSign(
...
+): boolean {
+  checkTxHash(Buffer.from(sessionKey.msg));
+  return musig.partialVerify({ sig, publicKey, publicNonce, sessionKey });
+}
+
+export function musig2AggregateSigs(sigs: Buffer[], sessionKey: SessionKey): Buffer {
+  return Buffer.from(musig.signAgg(sigs, sessionKey));
+}
+
+export function createMusig2SigningSession(sessionArgs: {
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/Musig2.ts:274]:
```suggestion
/** @return session key that can be used to reference the session later */
export function createMusig2SigningSession(sessionArgs: {
```

?

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
looks good overall

File: modules/utxo-lib/src/bitgo/UtxoPsbt.ts [hunk 2]
```diff
@@ -358,13 +349,84 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint> = UtxoTransaction<bigin
...
     const results = this.data.inputs.map((input, idx) => {
-      return input.tapScriptSig?.length
-        ? this.validateTaprootSignaturesOfInput(idx)
-        : this.validateSignaturesOfInput(idx, (p, m, s) => eccLib.verify(m, p, s));
+      if (input.tapScriptSig?.length) {
+        return this.validateTaprootSignaturesOfInput(idx);
+      } else if (input.partialSig?.length) {
+        return this.validateSignaturesOfInput(idx, (p, m, s) => eccLib.verify(m, p, s));
+      } else {
+        return this.validateTaprootMusig2SignaturesOfInput(idx);
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/UtxoPsbt.ts:357]: Done

File: modules/utxo-lib/src/bitgo/UtxoPsbt.ts [hunk 3]
```diff
@@ -358,13 +349,84 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint> = UtxoTransaction<bigin
...
+      }
     });
     return results.reduce((final, res) => res && final, true);
   }
 
+  validateTaprootMusig2SignaturesOfInput(inputIndex: number, pubkey?: Buffer): boolean {
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/UtxoPsbt.ts:372]: Done

File: modules/utxo-lib/src/bitgo/UtxoPsbt.ts [hunk 4]
```diff
@@ -358,13 +349,84 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint> = UtxoTransaction<bigin
...
+
+    const results: boolean[] = [];
+
+    mySigs.forEach((mySig) => {
+      const myNonce = nonces.find((kv) => kv.participantPubKey.equals(mySig.pubKey));
+      if (!myNonce) {
+        throw new Error('Found no pub nonce for pubkey');
+      }
+      results.push(musig2PartialSigVerify(mySig.sig, mySig.pubKey, myNonce.pubNonce, sessionKey));
+    });
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/UtxoPsbt.ts]: Done

File: modules/utxo-lib/src/bitgo/Musig2.ts (hunk 1 above)
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/Musig2.ts:275]: Done

**Review by OttoAllmendinger — APPROVED** [review]

**Pull request merged**